pkg/client/restclient/client_test.go

```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
+		t.Errorf("Expected object not to be created")
+	}
+	statusOut, err := runtime.Decode(testapi.Default.Codec(), body)
+	if err != nil {
+		t.Errorf("Unexpected error %#v", err)
+	}
+	if !reflect.DeepEqual(status, statusOut) {
+		t.Errorf("Unexpected mis-match. Expected %#v.  Saw %#v", status, statusOut)
+	}
+	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
+}
+
+func TestDoRequestAcceptedNoContentReturned(t *testing.T) {
+	status := &unversioned.Status{Status: unversioned.StatusSuccess}
+	expectedBody, _ := runtime.Encode(testapi.Default.Codec(), status)
+	fakeHandler := utiltesting.FakeHandler{
+		StatusCode:   204,
+		ResponseBody: string(expectedBody),
+		T:            t,
+	}
+	testServer := httptest.NewServer(&fakeHandler)
+	defer testServer.Close()
+	c, err := RESTClientFor(&Config{
+		Host: testServer.URL,
+		ContentConfig: ContentConfig{
+			GroupVersion:         testapi.Default.GroupVersion(),
+			NegotiatedSerializer: testapi.Default.NegotiatedSerializer(),
+		},
+		Username: "user",
+		Password: "pass",
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	created := false
+	body, err := c.Get().Prefix("test").Do().WasCreated(&created).Raw()
+	if err != nil {
+		t.Errorf("Unexpected error %#v", err)
+	}
+	if created {
+		t.Errorf("Expected object not to be created")
+	}
+	statusOut, err := runtime.Decode(testapi.Default.Codec(), body)
+	if err == nil {
+		t.Errorf("Expected Error")
+	}
+	if statusOut != nil {
+		t.Errorf("Unexpected mis-match. Expected %#v.  Saw %#v", nil, statusOut)
+	}
+	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
+}
+
+func TestBadRequest(t *testing.T) {
+	status := &unversioned.Status{Status: fmt.Sprintf("%s", unversioned.StatusReasonBadRequest)}
+	expectedBody, _ := runtime.Encode(testapi.Default.Codec(), status)
+	fakeHandler := utiltesting.FakeHandler{
+		StatusCode:   400,
+		ResponseBody: string(expectedBody),
+		T:            t,
+	}
+	testServer := httptest.NewServer(&fakeHandler)
+	defer testServer.Close()
+	c, err := RESTClientFor(&Config{
+		Host: testServer.URL,
+		ContentConfig: ContentConfig{
+			GroupVersion:         testapi.Default.GroupVersion(),
+			NegotiatedSerializer: testapi.Default.NegotiatedSerializer(),
+		},
+		Username: "user",
+		Password: "pass",
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	created := false
+	body, err := c.Get().Prefix("test").Do().WasCreated(&created).Raw()
+	if err == nil {
+		t.Errorf("Expected error")
+	}
+	if created {
+		t.Errorf("Expected object not to be created")
+	}
+	statusOut, err := runtime.Decode(testapi.Default.Codec(), body)
+	if err == nil {
+		t.Errorf("Expected Error")
+	}
+	if statusOut != nil {
+		t.Errorf("Unexpected mis-match. Expected %#v.  Saw %#v", nil, statusOut)
+	}
+	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
+}
+
+func TestHttpMethods(t *testing.T) {
```

Comment:
This is unnecessary.  If these are ever nil the controller and client tests will just fall over